lib/components/ImageButton.tsx

```diff
@@ -0,0 +1,226 @@
+/**
+ * @file
+ * @copyright 2024 Aylong (https://github.com/AyIong)
+ * @license MIT
+ */
+
+import { Placement } from '@popperjs/core';
+import { ReactNode } from 'react';
+
+import { BooleanLike, classes } from '../common/react';
+import styles from '../styles/components/ImageButton.module.scss';
+import { BoxProps, computeBoxProps } from './Box';
+import { DmIcon } from './DmIcon';
+import { Icon } from './Icon';
+import { Image } from './Image';
+import { Stack } from './Stack';
+import { Tooltip } from './Tooltip';
+
+type Props = Partial<{
+  /** Asset cache. Example: `asset={['assetname32x32', thing.key]}` */
+  asset: string[];
+  /** Classic way to put images. Example: `base64={thing.image}` */
+  base64: string;
+  /** 
+   * Special container for buttons.
+   * You can put any other component here.
+   * Has some special stylings!
+   * Example: `buttons={<Button>Send</Button>}` 
+   */
+  buttons: ReactNode;
+  /** Enables alternate buttons container. */
```

Comment:
What does this enable? `buttons`? It's a bit unclear

Reply:
insofar as the 'alternative layout' doesn't have any explanation on what it is

Reply:
I hope I've explained it informatively enough, I don't want to stretch out the documentation